src/k2/module/k23si/Module.cpp

```diff
@@ -1129,6 +1204,7 @@ K23SIPartitionModule::_createWI(dto::K23SIWriteRequest&& request, VersionSet& ve
 
     // the TWIM accepted the write. Add it as a WI now
     versions.WI.emplace(std::move(rec), request.request_id);
+    _activeWI++;
```

Comment:
I don't quite like the management of this count along with the std::optional. Is there some way we can tie the counter to the lifetime of the object?

Reply:
Discussed with Justin and we decided to use the _removeWI function in all places to maintain consistency. Just to clarify, the _activeWI keeps track of number of activeWI over all keys.

Reply:
Decided to introduce new metric to keep track of finalizedWI and change activeWI (gauge) metric to totalWI (counter).

Reply:
done